lib/nostrum/shard.ex

```diff
@@ -84,6 +87,12 @@ defmodule Nostrum.Shard do
   def ondisconnect(reason, state) do
     Logger.warn "WS DISCONNECTED BECAUSE: #{inspect reason}"
     Logger.debug "STATE ON CLOSE: #{inspect state}"
+
+    if state.heartbeat_task do
+      Task.shutdown(state.heartbeat_task)
+      state = %{state | heartbeat_task: nil}
```

Comment:
It's considered unsafe (read: bad practice) to set a variable inside of `if`, `cond`, `case`, etc. Consider rewriting this to something like 
```Elixir
state = 
  if state.heartbeat_task do
    Task.shutdown(state.heartbeat_task)
    %{state | heartbeat_task: nil}
  end
```